datafusion/spark/src/function/math/abs.rs

```diff
@@ -0,0 +1,1136 @@
+                            ScalarValue::Decimal256(Some(abs_val), *precision, *scale),
+                        )),
+                        None => {
+                            if !fail_on_error {
+                                // return the original value
+                                Ok(ColumnarValue::Scalar(ScalarValue::Decimal256(
+                                    Some(v),
+                                    *precision,
+                                    *scale,
+                                )))
+                            } else {
+                                Err(arithmetic_overflow_error("Decimal256"))
+                            }
+                        }
+                    })
+                    .unwrap(),
+                ScalarValue::IntervalYearMonth(a) => {
+                    let result = a.map(|v| v.wrapping_abs());
+                    Ok(ColumnarValue::Scalar(ScalarValue::IntervalYearMonth(
+                        result,
+                    )))
+                }
+                ScalarValue::IntervalDayTime(a) => {
+                    let result = a.map(|v| v.wrapping_abs());
+                    Ok(ColumnarValue::Scalar(ScalarValue::IntervalDayTime(result)))
+                }
+
+                dt => internal_err!("Not supported datatype for Spark ABS: {dt}"),
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datafusion_common::cast::{
+        as_decimal128_array, as_decimal256_array, as_float32_array, as_float64_array,
+        as_int16_array, as_int32_array, as_int64_array, as_int8_array,
+        as_interval_dt_array, as_interval_ym_array, as_uint64_array,
+    };
+
+    fn with_fail_on_error<F: Fn(bool) -> Result<()>>(test_fn: F) {
+        for fail_on_error in [true, false] {
+            let _ = test_fn(fail_on_error);
+        }
+    }
+
+    // Unsigned types, return as is
+    #[test]
+    fn test_abs_u8_scalar() {
+        with_fail_on_error(|fail_on_error| {
+            let args = ColumnarValue::Scalar(ScalarValue::UInt8(Some(u8::MAX)));
+            let fail_on_error_arg =
+                ColumnarValue::Scalar(ScalarValue::Boolean(Some(fail_on_error)));
+            match spark_abs(&[args, fail_on_error_arg]) {
+                Ok(ColumnarValue::Scalar(ScalarValue::UInt8(Some(result)))) => {
+                    assert_eq!(result, u8::MAX);
+                    Ok(())
+                }
+                Err(e) => {
+                    if fail_on_error {
+                        assert!(
+                            e.to_string().contains("ARITHMETIC_OVERFLOW"),
+                            "Error message did not match. Actual message: {e}"
+                        );
+                        Ok(())
+                    } else {
+                        panic!("Didn't expect error, but got: {e:?}")
+                    }
+                }
+                _ => unreachable!(),
+            }
+        });
+    }
```

Comment:
This test design is very confusing; we can't tell if a test case is meant to return Ok or Err as it automatically does the "correct" verification for each case. This automatic way of passing the test on Err should be switched so if we have a test case that is meant to return Err, that is the only thing we check for.